official/utils/logs/hooks.py

```diff
@@ -73,14 +73,30 @@ def __init__(self,
     self._total_steps = 0
     self._batch_size = batch_size
     self._warm_steps = warm_steps
+    self._train_op = None
+
+  def specify_train_op(self, train_op):
+    self._train_op = train_op
 
   def begin(self):
     """Called once before using the session to check global step."""
-    self._global_step_tensor = tf.train.get_global_step()
-    if self._global_step_tensor is None:
+    if self._train_op is None:
```

Comment:
Since you have a if check for estimator.train_op is None below, it seems that it is possible for estimator to not have a train_op. Should we handle it more gracefully, eg, backoff to the original behavior if the train_op is not specified.